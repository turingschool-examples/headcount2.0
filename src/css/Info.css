.info-modal {
  position: absolute;
  width: 600px;
  height: 200px;
  right: 0;
  top: 150px;
  background: #fff;
  box-shadow: 2px 2px 2px rgba(0, 0, 0, .2);
  border: 2px #000 solid;
  border-right: 0;
  opacity: 1;
  border-top-left-radius: 10px;
  border-bottom-left-radius: 10px;
  transition: .3s;
}

.info-modal h1 {
  padding-top: 0;
  text-align: center;
}

.info-modal h1,
.info-modal p {
  margin: 1rem;
}

.info-modal img {
  margin: 1rem 0 0 1rem;
  width: 25px;
  transition: .2s;
}

.info-modal img:hover {
  cursor: pointer;
  transform: rotate(90deg);
}

.hidden-modal {
  opacity: 0;
  right: -80%;
}