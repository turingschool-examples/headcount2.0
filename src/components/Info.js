import React from 'react';
import PropTypes from 'prop-types';

import '../css/Info.css';

const Info = (props) => {
  return (
    <aside className={props.modalClass}>
      <img 
        className='close-icon'
        src='./cancel.svg' 
        alt='close-icon'
        onClick={() => {
          props.untoggleModal();
        }} 
      />
      <h1>About Colorado Headcount</h1>
      <p>This app allows you to compare kindergarten test scores between the school districts throughout Colorado. To compare two districts, simply click their cards to select them.</p>
    </aside>
  );
};

Info.propTypes = {
  modalClass: PropTypes.string.isRequired,
  untoggleModal: PropTypes.func.isRequired
};

export default Info;