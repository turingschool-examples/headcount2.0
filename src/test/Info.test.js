import React from 'react';
import { shallow } from 'enzyme';
import Info from '../components/Info';


describe('Info', ()=>{
  let wrapper; 

  const mockClass = 'hidden-modal info-modal';

  const mockUntoggleModal = jest.fn();

  beforeEach(()=>{
    wrapper = shallow(<Info 
      modalClass={mockClass}
      untoggleModal={mockUntoggleModal}
    />);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should render content to the DOM', () => {
    expect(wrapper).toMatchSnapshot();    
  });

  it('should call untoggleModal on button click', () => {
    wrapper.find('.close-icon').simulate('click');

    expect(mockUntoggleModal.mock.calls.length).toBe(1);
  });

});