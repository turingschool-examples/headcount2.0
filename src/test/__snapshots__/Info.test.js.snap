// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info should render content to the DOM 1`] = `
<aside
  aria-label="info-section"
  className="hidden-modal info-modal"
>
  <img
    alt="close-icon"
    className="close-icon"
    onClick={[Function]}
    src="./cancel.svg"
  />
  <h1
    aria-label="info-header-label"
  >
    About Colorado Kindercount
  </h1>
  <p
    aria-label="info-description"
  >
    This app allows you to compare kindergarten test scores between the school districts throughout Colorado. To compare two districts, simply click their cards to select them.
  </p>
</aside>
`;
